aten/src/ATen/BatchedTensorImpl.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <bitset>
+
+#include <ATen/ArrayRef.h>
+#include <ATen/SmallVector.h>
+#include <ATen/Tensor.h>
+
+namespace at {
+
+// We assume this in a few other places in the codebase,
+// but there isn't a centralized definition.
+constexpr int64_t kVmapMaxTensorDims = 64;
+
+struct BatchDim {
+  BatchDim(int64_t dim, int64_t level) : dim_(dim), level_(level) {}
+  int64_t dim() const {
+    return dim_;
+  }
+  int64_t level() const {
+    return level_;
+  }
+ private:
+  int64_t dim_;
+  int64_t level_;
+};
+
+using BatchDims = SmallVector<BatchDim, 5>;
```

Comment:
i would push '5' as a constant next to `kVmapMaxTensorDims`. Also, why 5?

Reply:
It's a hypothesis that users won't use more than 5 batch dimensions. In PyTorch we guarantee "best" performance on tensors of dimensionality less than or equal to 5 (TensorImpl's sizes/strides use `SmallVector<int, 5>`).

I'll toss the number into a constant as you suggested (y)